src-electron/generator/helper-zcl.js

```diff
@@ -2456,6 +2456,172 @@ async function format_zcl_string_as_characters_for_generated_defaults(
+        context.global.db,
+        packageIds,
+        dataType.name
+      )
+      dataTypesize = Math.pow(2, Math.ceil(Math.log2(number.size)))
+    }
+  } else {
+    env.logError(`Data Type ${type} is not defined in the data type table.`)
+  }
+  return {
+    isTypeSigned: isTypeSigned,
+    dataTypesize: dataTypesize,
+    dataType: dataType,
+  }
+}
+
+/**
+ * Given a zcl data type return the min allowed value for that zcl data type
+ * based on the language specified in the options
+ * @param {*} type
+ * @param {*} options
+ * @returns max allowed value for the given zcl data type
+ * Available Options:
+ * - language: determines the output of the helper based on language
+ * for eg: (as_type_max_value language='c++') will give the output specific to
+ * the c++ language.
+ */
+async function as_type_min_value(type, options) {
+  let signAndSize = await get_sign_and_size_of_zcl_type(type, this)
+  let isTypeSigned = signAndSize.isTypeSigned
+  let dataTypesize = signAndSize.dataTypesize
+  let dataType = signAndSize.dataType
+
+  // Returning min value based on c/c++ language
+  if (options.hash.language == 'c++') {
+    if (!isTypeSigned) {
+      // Exceptions: single and double zcl types
+      if (dataType && dataType.name == 'single') {
+        return `-std::numeric_limits<float>::infinity()`
+      } else if (dataType && dataType.name == 'double') {
+        return `-std::numeric_limits<double>::infinity()`
+      } else {
+        return 0
+      }
+    } else {
+      return 'INT' + dataTypesize * 8 + '_MIN'
+    }
+  } else {
+    return dataTypesize * 8
+  }
+}
+
+/**
+ * Given a zcl data type return the max allowed value for that zcl data type
+ * based on the language specified in the options
+ * @param {*} type
+ * @param {*} options
+ * @returns max allowed value for the given zcl data type
+ * Available Options:
+ * - language: determines the output of the helper based on language
+ * for eg: (as_type_max_value language='c++') will give the output specific to
+ * the c++ language.
+ */
+async function as_type_max_value(type, options) {
+  let signAndSize = await get_sign_and_size_of_zcl_type(type, this)
+  let isTypeSigned = signAndSize.isTypeSigned
+  let dataTypesize = signAndSize.dataTypesize
+  let dataType = signAndSize.dataType
+
+  // Returning max value based on c++ language
+  if (options.hash.language == 'c++') {
+    if (!isTypeSigned) {
+      // Exceptions: single, double and boolean zcl types
+      if (dataType && dataType.name == 'single') {
+        return `std::numeric_limits<float>::infinity()`
+      } else if (dataType && dataType.name == 'double') {
+        return `std::numeric_limits<double>::infinity()`
+      } else if (dataType && dataType.name == 'boolean') {
+        return `1`
+      } else {
+        return 'UINT' + dataTypesize * 8 + '_MAX'
+      }
+    } else {
+      return 'INT' + dataTypesize * 8 + '_MAX'
+    }
+  } else {
+    return dataTypesize * 8
```

Comment:
It felt useful to just give the size away as a default when no language is specified. They could use this size in their template in the future. May be I should mention size is in bits or even give an option if a user wants the size in bytes or bits

Reply:
I don't see how it can be useful.  I ask for the min value for int64u, I get back "64".  I ask for the max value, I also get "64".  How is that useful?

Reply:
That is what I was trying to say. for max you get back 64 which can be used in future templates but min can throw an error.

Reply:
Ah, I see.  But for an `int64s`, what does "max is 64" tell you that is actionable?  How would you use it?

Reply:
So in the template a user could do something like `int{{as_type_max_value type}}s` or `int{{as_type_max_value type}}` instead of `{{as_type_max_value type language='c++'}}` despite no specific language mentioned. The template can use that max size in different use cases that may arise.

Reply:
Well, it does not have to return the value. I can throw an error here too. We can have a separate helper for size if need be to avoid this confusion. This was more of a way to get size than max.

Reply:
Separate helper for size would be _much_ clearer.